openassessment/assessment/api/staff.py

```diff
@@ -0,0 +1,292 @@
+"""
+Public interface for staff grading, used by students/course staff.
+"""
+import logging
+from django.db import DatabaseError, IntegrityError, transaction
+from dogapi import dog_stats_api
+
+from openassessment.assessment.models import (
+    Assessment, AssessmentFeedback, AssessmentPart,
+    InvalidRubricSelection
+)
+from openassessment.assessment.serializers import (
+    AssessmentFeedbackSerializer, RubricSerializer,
+    full_assessment_dict, rubric_from_dict, serialize_assessments,
+    InvalidRubric
+)
+from openassessment.assessment.errors import (
+    StaffAssessmentRequestError, StaffAssessmentInternalError
+)
+from submissions import api as sub_api
+
+logger = logging.getLogger("openassessment.assessment.api.staff")
+
+
+STAFF_TYPE = "ST"
+
+
+def submitter_is_finished(submission_uuid, requirements):
+    """
+    Determine if the submitter has finished their requirements for staff
+    assessment. Always returns True.
+
+    Args:
+        submission_uuid (str): Not used.
+        requirements (dict): Not used.
+
+    Returns:
+        True
+
+    """
+    return True
+
+
+def assessment_is_finished(submission_uuid, requirements):
+    """
+    Determine if the assessment of the given submission is completed. This
+    checks to see if staff have completed the assessment.
+
+    Args:
+        submission_uuid (str): The UUID of the submission being graded.
+        requirements (dict): Any variables that may effect this state.
+
+    Returns:
+        True if the assessment has been completed for this submission.
+
+    """
+    if requirements and requirements.get('staff', {}).get('required', False):
```

Comment:
This logic is a little subtle and I'm not sure I fully understand it. 

I believe this is saying that there are two cases:
1. If the problem requires staff grading, then the assessment is only true if an assessment exists, which must by definition be staff graded. I'm not sure if there is a risk that a score was provided by some other means before this problem was made to require staff grading. Maybe it would be better to confirm that the latest assessment is of type staff assessment.
2. If the problem does not require staff grading, then the submission is always considered finished. This seems counter-intuitive to me, because if the problem doesn't require staff grading then doesn't it need to wait for some other kind of assessment? Is it the case that then this code wouldn't be invoked, because it isn't a staff grading workflow, but would instead go to peer or something else? If that's the case, maybe it would be better to throw an exception than to always return True.